common/heroku.go

```diff
@@ -43,28 +49,33 @@ func makeRequest(method, url, token string, body io.Reader) (req *http.Request,
 	return req, nil
 }
 
-func handleRequest(req *http.Request, f interface{}) error {
+func handleRequest(req *http.Request) error {
 	httpClient := &http.Client{}
 	resp, err := httpClient.Do(req)
 	if err != nil {
 		log.Println("could not make http request: ", err.Error())
 		return err
 	}
-	return parseResponse(resp, f)
+	return parseResponse(resp)
 }
 
-func parseResponse(resp *http.Response, f interface{}) error {
+func parseResponse(resp *http.Response) error {
 	defer resp.Body.Close()
-	responseAsBytes, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		log.Println("could not read resp body", err.Error())
-		return err
-	}
 
-	err = json.Unmarshal(responseAsBytes, &f)
-	if err != nil {
-		log.Println("could not parse response: ", err.Error())
-		return NewError("could not parse : %s", string(responseAsBytes))
+	if resp.StatusCode != 200 {
```

Comment:
Just wanted to clarify if the intent here is to execute the original code process, if the status code detected is not ok(!200)?

Reply:
My thought here is that the calling function doesn't really care about Heroku's response IF it succeeded... it only cares if there is an error. When I tested the function, Heroku returns a 200 response if it successfully updated the env vars with the response body containing all the keys and values.

So I added the checking that if the response status code is not 200 (there's an error/issue), then it will parse the response to get the error message and return that directly to the calling function. If the update is successful, it will just return error = nil.

Reply:
Got it! Thanks for the clarification!